### PR TITLE
Dts 26342 daily scrum Screen 1

### PR DESCRIPTION
**Pull request by Chittauri**

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+				LinkedHashMap<String, StandUpViewKpiData> cardDetails = new LinkedHashMap<>();
+				String role = UNASSIGNED;
+				if (MapUtils.isNotEmpty(userWiseRole)) {
+					role = userWiseRole.getOrDefault(listEntry.getKey(), UNASSIGNED);
+				}
+				List<JiraIssue> jiraIssueList = listEntry.getValue();
+				String assigneeId = listEntry.getKey();
+				String assigneeName = jiraIssueList.stream().findFirst().orElse(new JiraIssue()).getAssigneeName();
+
+				cardDetails.put("Remaining Capacity", userWiseRemainingCapacity.getOrDefault(assigneeId,
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java:253]: Use Constants for String Values

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+					String.valueOf(totalEstimate), null, estimationUnit));
+
+			int totalRemainingEstimate = jiraIssueList.stream()
+					.mapToInt(issue -> Optional.ofNullable(issue.getRemainingEstimateMinutes()).orElse(0)).sum();
+			assigneeWiseRemainingEstimate.put(assigneeId,
+					new StandUpViewKpiData(String.valueOf(totalRemainingEstimate), null, CommonConstant.DAY, null));
+		});
+	}
+
+	private void calculateRemainigCapacity(int daysBetween, int daysLeft, AssigneeCapacity assignee,
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java:345]: Comments and Documentation

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+import com.publicissapient.kpidashboard.apis.jira.service.JiraKPIService;
+import com.publicissapient.kpidashboard.apis.model.Filter;
+import com.publicissapient.kpidashboard.apis.model.KpiElement;
+import com.publicissapient.kpidashboard.apis.model.KpiRequest;
+import com.publicissapient.kpidashboard.apis.model.Node;
+import com.publicissapient.kpidashboard.apis.model.TreeAggregatorDetail;
+import com.publicissapient.kpidashboard.apis.util.CommonUtils;
+import com.publicissapient.kpidashboard.apis.util.KpiDataHelper;
+import com.publicissapient.kpidashboard.common.constant.CommonConstant;
+import com.publicissapient.kpidashboard.common.model.application.AssigneeCapacity;
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java:59]: Remove unused and Optimize Imports

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+	private void calculateAssigneeWiseMaxDelay(List<IterationPotentialDelay> maxPotentialDelayList,
+			Map<String, StandUpViewKpiData> assigneeWiseMaxDelay) {
+		Map<String, List<IterationPotentialDelay>> collect = maxPotentialDelayList.stream()
+				.filter(IterationPotentialDelay::isMaxMarker)
+				.collect(Collectors.groupingBy(IterationPotentialDelay::getAssigneeId));
+
+		collect.forEach((assigneeId, delayList) -> {
+			int totalDelayInMinutes = (int)delayList.stream()
+					.mapToDouble(delay -> Optional.of(delay.getPotentialDelay()).orElse(0) * 60 * 8).sum();
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java:316]: Avoid Nested Optional

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+			assigneeWiseMaxDelay.put(assigneeId,
+					new StandUpViewKpiData(String.valueOf(totalDelayInMinutes), null, CommonConstant.DAY, null));
+		});
+	}
+
+	private void calculateAssigneeWiseRemainingEstimate(Map<String, List<JiraIssue>> assigneeWiseNotCompleted,
+			Map<String, StandUpViewKpiData> remainingWork,
+			Map<String, StandUpViewKpiData> assigneeWiseRemainingEstimate, String estimationCriteria) {
+		String estimationUnit = estimationCriteria.equalsIgnoreCase(CommonConstant.STORY_POINT) ? CommonConstant.SP
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DailyStandupServiceImpl.java:326]: Use Method References lambda

**Review by hirbabar — APPROVED** [review]

**Review by BhagyashreeShettigar — APPROVED** [review]

**Pull request merged**